src/cherrypick.ts

```diff
@@ -130,23 +130,17 @@ export class CherryPick {
 
       let commitShasToCherryPick: string[];
 
-      if (mainpr.commits == 1) {
-        // if the Pr only has one commit, we don't care
-        // if the Pr was squashed or rebased
-        commitShasToCherryPick = commitShas;
+      // find out if "squashed and merged" or "rebased and merged"
+      if (await this.github.isSquashed(mainpr)) {
```

Comment:
The logic would go like this I guess

```suggestion
      // find out if "squashed and merged" or "rebased and merged"
      if (await this.github.isSquashed(mainpr) or main.commits == 1) {
```

Reply:
check #7 @windmgc

Reply:
Ah, that makes sense. The alternative PR looks good to me!